stdlib/LibGit2/test/libgit2.jl

```diff
@@ -11,11 +11,15 @@ isdefined(Main, :TestHelpers) || @eval Main include(joinpath($(BASE_TEST_PATH),
 import .Main.TestHelpers: with_fake_pty
 
 function challenge_prompt(code::Expr, challenges; timeout::Integer=60, debug::Bool=true)
+    input_code = tempname()
+    open(input_code, "w") do fp
+        serialize(fp, code)
+    end
     output_file = tempname()
     wrapped_code = quote
         using Serialization
-        result = let
-            $code
+        result = open($input_code) do fp
+            eval(deserialize(fp))
```

Comment:
Why is this change now necessary?

Reply:
I'm not sure if it's still necessary, but it definitely was at one point and it seems like a good improvement.  Before we were trusting that the string representation of a expression was round-trippable.  This isn't just `Expr`s, it also includes embedded credential objects spliced directly into the AST.  If their stringification isn't round-trippable, this fails.